src/coverlet.collector/DataCollection/CoverletCoverageCollector.cs

```diff
@@ -130,12 +130,11 @@ private void OnSessionEnd(object sender, SessionEndEventArgs e)
                 _eqtTrace.Verbose("{0}: SessionEnd received", CoverletConstants.DataCollectorName);
 
                 // Get coverage reports
-                string coverageReport = _coverageManager?.GetCoverageReport();
+                IEnumerable<(string report, string fileName)> coverageReports = _coverageManager?.GetCoverageReports();
 
                 // Send result attachments to test platform.
-                var attachmentManager = new AttachmentManager(_dataSink, _dataCollectionContext, _logger, _eqtTrace, this.GetReportFileName());
-                attachmentManager?.SendCoverageReport(coverageReport);
-
+                var attachmentManager = new AttachmentManager(_dataSink, _dataCollectionContext, _logger, _eqtTrace);
+                coverageReports?.ToList().ForEach(report => attachmentManager.SendCoverageReport(report.report, report.fileName));
```

Comment:
we need to move attachment manager creation inside foreach because on disposing we delete folder and if we've more than one format we get exception for access denied(first format remove folder). I did some test...i think something like
```
Starting test execution, please wait...
Data collector 'XPlat code coverage' message: Coverlet.Collector.Utilities.CoverletDataCollectorException: CoverletCoverageDataCollector: Failed to cleanup report directory: 'C:\Users\Marco\AppData\Local\Temp\eeff0599-06e6-434c-8f47-d95333cf66f4' ---> System.UnauthorizedAccessException: Access to the path 'coverage.json' is denied.
   at System.IO.Win32FileSystem.RemoveDirectoryHelper(String fullPath, Boolean recursive, Boolean throwOnTopLevelDirectoryNotFound)
   at System.IO.Win32FileSystem.RemoveDirectory(String fullPath, Boolean recursive)
   at System.IO.Directory.Delete(String path, Boolean recursive)
   at Coverlet.Collector.Utilities.DirectoryHelper.Delete(String path, Boolean recursive) in C:\git\coverlet\src\coverlet.collector\Utilities\DirectoryHelper.cs:line 24
   at Coverlet.Collector.DataCollection.AttachmentManager.CleanupReportDirectory() in C:\git\coverlet\src\coverlet.collector\DataCollection\AttachmentManager.cs:line 156
   --- End of inner exception stack trace ---
   at Coverlet.Collector.DataCollection.AttachmentManager.CleanupReportDirectory() in C:\git\coverlet\src\coverlet.collector\DataCollection\AttachmentManager.cs:line 163
   at Coverlet.Collector.DataCollection.AttachmentManager.OnSendFileCompleted(Object sender, AsyncCompletedEventArgs e) in C:\git\coverlet\src\coverlet.collector\DataCollection\AttachmentManager.cs:line 120.

Attachments:
  C:\git\coverletissue\collectorlog\XUnitTestProject1\TestResults\451d44d4-5d77-432a-a53b-ec1c472cbdda\coverage.json
  C:\git\coverletissue\collectorlog\XUnitTestProject1\TestResults\451d44d4-5d77-432a-a53b-ec1c472cbdda\coverage.cobertura.xml
```
```cs
 coverageReports?.ToList().ForEach(report =>
                {
                    // Send result attachments to test platform.
                    var attachmentManager = new AttachmentManager(_dataSink, _dataCollectionContext, _logger, _eqtTrace);
                    attachmentManager.SendCoverageReport(report.report, report.fileName);
                });
```
and 
```
C:\git\coverletissue\collectorlog
λ dotnet test --collect:"XPlat Code Coverage" --settings runsettings XUnitTestProject1\
Test run for C:\git\coverletissue\collectorlog\XUnitTestProject1\bin\Debug\netcoreapp2.2\XUnitTestProject1.dll(.NETCoreApp,Version=v2.2)
Microsoft (R) Test Execution Command Line Tool Version 16.2.0-preview-20190606-02
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...

Attachments:
  C:\git\coverletissue\collectorlog\XUnitTestProject1\TestResults\85149cf7-1ea5-4223-a823-f6cb7781fbb5\coverage.json
  C:\git\coverletissue\collectorlog\XUnitTestProject1\TestResults\85149cf7-1ea5-4223-a823-f6cb7781fbb5\coverage.cobertura.xml

Test Run Successful.
Total tests: 1
     Passed: 1
 Total time: 1,8736 Seconds
```
runsetting sample
```xml
<?xml version="1.0" encoding="utf-8" ?>
<RunSettings>
  <DataCollectionRunSettings>
    <DataCollectors>
      <DataCollector friendlyName="XPlat code coverage">
        <Configuration>
          <Format>json,cobertura</Format>
        </Configuration>
      </DataCollector>
    </DataCollectors>
  </DataCollectionRunSettings>
</RunSettings>
```
@daveMueller double check with me and let me know!

Reply:
Yes I can confirm that. Sorry that I missed this in my initial commit. I adapted the code.